src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
 
 /**
- * The chat window component.  Displays text messages and
- * accepts input.  Also acts as a {@link CommandEncoder},
- * encoding/decoding commands that display message in the text area
+ * The chat window component. Displays text messages and accepts input. Also
+ * acts as a {@link CommandEncoder}, encoding/decoding commands that display
+ * message in the text area
  */
 public class Chatter extends JPanel implements CommandEncoder, Buildable {
   private static final long serialVersionUID = 1L;
 
-  protected JTextArea conversation;
-  protected JTextField input;
-  protected JScrollPane scroll = new ScrollPane(
-       JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
-       JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
-  protected static final String MY_CHAT_COLOR = "myChatColor"; //$NON-NLS-1$
-  protected static final String OTHER_CHAT_COLOR = "otherChatColor"; //$NON-NLS-1$
-  protected static final String GAME_MSG_COLOR = "gameMessageColor"; //$NON-NLS-1$
-  protected static final String SYS_MSG_COLOR = "systemMessageColor"; //$NON-NLS-1$
-
-  protected Color gameMsg;
-  protected Color systemMsg;
-  protected Color myChat;
-  protected Color otherChat;
+  protected JTextPane conversation;
+  protected HTMLDocument doc;
+  protected HTMLEditorKit kit;
+  protected StyleSheet style;
 
+  protected JTextField input;
+  protected JScrollPane scroll = new ScrollPane(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+  protected JScrollPane scroll2 = new ScrollPane(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+  protected static final String MY_CHAT_COLOR    = "HTMLChatColor";          //$NON-NLS-1$ //BR// Different tags to "restart" w/ new default scheme
+  protected static final String OTHER_CHAT_COLOR = "HTMLotherChatColor";     //$NON-NLS-1$
+  protected static final String GAME_MSG1_COLOR  = "HTMLgameMessage1Color";  //$NON-NLS-1$
+  protected static final String GAME_MSG2_COLOR  = "HTMLgameMessage2Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG3_COLOR  = "HTMLgameMessage3Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG4_COLOR  = "HTMLgameMessage4Color";  //$NON-NLS-1$ //BR//
+  protected static final String GAME_MSG5_COLOR  = "HTMLgameMessage5Color";  //$NON-NLS-1$ //BR//
+  protected static final String SYS_MSG_COLOR    = "HTMLsystemMessageColor"; //$NON-NLS-1$
+  protected static final String HTML_COMPATIBILITY = "HTMLcompatibility";    //$NON-NLS-1$
+
+  protected Font myFont;
+
+  protected Color gameMsg, gameMsg2, gameMsg3, gameMsg4, gameMsg5;
+  protected Color systemMsg, myChat, otherChat;
+  
+  protected boolean htmlCompatibility; //BR// Disable HTML parsing unless opted in, to maintain backwards compatibility
+	
   public static String getAnonymousUserName() {
     return Resources.getString("Chat.anonymous"); //$NON-NLS-1$
   }
 
-  public Chatter() {
+  public Chatter() {		
```

Comment:
Trailing whitespace, and also a piece of code that shows up in the diff even though nothing was changed here

Reply:
fixed (hopefully)